[exporter/datadog] Flaky TestTraceExporter

**Describe the bug**

The `TestTraceExporter` is flaky.

**Steps to reproduce**

Be unlucky and hit this on CI :)

**What did you expect to see?**

Test pass reliably

**What did you see instead?**

Test sometimes fails depending on timing:

```
2022-06-30T11:20:36.0260825Z === RUN   TestTraceExporter
2022-06-30T11:20:36.0261295Z     traces_exporter_test.go:276: 
2022-06-30T11:20:36.0261829Z         	Error Trace:	traces_exporter_test.go:276
2022-06-30T11:20:36.0262279Z         	Error:      	Not equal: 
2022-06-30T11:20:36.0262992Z         	            	expected: "application/x-protobuf"
2022-06-30T11:20:36.0263515Z         	            	actual  : ""
2022-06-30T11:20:36.0263908Z         	            	
2022-06-30T11:20:36.0264323Z         	            	Diff:
2022-06-30T11:20:36.0264850Z         	            	--- Expected
2022-06-30T11:20:36.0265312Z         	            	+++ Actual
2022-06-30T11:20:36.0265824Z         	            	@@ -1 +1 @@
2022-06-30T11:20:36.0266438Z         	            	-application/x-protobuf
2022-06-30T11:20:36.0266862Z         	            	+
2022-06-30T11:20:36.0267291Z         	Test:       	TestTraceExporter
2022-06-30T11:20:36.0267726Z --- FAIL: TestTraceExporter (0.20s)
```

**Additional context**

@gbbr will work on solving this

--- a/exporter/datadogexporter/config.go
+++ b/exporter/datadogexporter/config.go
@@ -266,6 +266,10 @@ type TracesConfig struct {
 	// If set to false the resource name will be filled with the instrumentation library name + span kind.
 	// The default value is `false`.
 	SpanNameAsResourceName bool `mapstructure:"span_name_as_resource_name"`
+
+	// flushInterval defines the interval in seconds at which the writer flushes traces
+	// to the intake; used in tests.
+	flushInterval float64
 }
 
 // TagsConfig defines the tag-related configuration

--- a/exporter/datadogexporter/traces_exporter.go
+++ b/exporter/datadogexporter/traces_exporter.go
@@ -74,6 +74,9 @@ func newTracesExporter(ctx context.Context, params component.ExporterCreateSetti
 	acfg.Endpoints[0].APIKey = cfg.API.Key
 	acfg.Ignore["resource"] = cfg.Traces.IgnoreResources
 	acfg.ReceiverPort = 0 // disable HTTP receiver
+	if v := cfg.Traces.flushInterval; v > 0 {
+		acfg.TraceWriter.FlushPeriodSeconds = v
+	}
 	if addr := cfg.Traces.Endpoint; addr != "" {
 		acfg.Endpoints[0].Host = addr
 	}

--- a/exporter/datadogexporter/traces_exporter_test.go
+++ b/exporter/datadogexporter/traces_exporter_test.go
@@ -231,10 +231,10 @@ func TestTraceExporter(t *testing.T) {
 	metricsServer := testutils.DatadogServerMock()
 	defer metricsServer.Close()
 
-	var got string
+	got := make(chan string, 1)
 	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		assert.Equal(t, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", req.Header.Get("DD-Api-Key"))
-		got = req.Header.Get("Content-Type")
+		got <- req.Header.Get("Content-Type")
 		rw.WriteHeader(http.StatusAccepted)
 	}))
 
@@ -259,6 +259,7 @@ func TestTraceExporter(t *testing.T) {
 				Endpoint: server.URL,
 			},
 			IgnoreResources: []string{},
+			flushInterval:   0.1,
 		},
 	}
 
@@ -270,9 +271,14 @@ func TestTraceExporter(t *testing.T) {
 	ctx := context.Background()
 	err = exporter.ConsumeTraces(ctx, simpleTraces())
 	assert.NoError(t, err)
-	time.Sleep(10 * time.Millisecond) // we need a bit of time for channels to receive things
+	timeout := time.After(2 * time.Second)
+	select {
+	case out := <-got:
+		require.Equal(t, "application/x-protobuf", out)
+	case <-timeout:
+		t.Fatal("Timed out")
+	}
 	require.NoError(t, exporter.Shutdown(context.Background()))
-	require.Equal(t, "application/x-protobuf", got)
 }
 
 func TestNewTracesExporter(t *testing.T) {

